RadixWallet/Features/DappInteractionFeature/Children/AccountsProofOfOwnership/AccountsProofOfOwnership.swift

```diff
@@ -0,0 +1,137 @@
+// MARK: - AccountsProofOfOwnership
+@Reducer
+struct AccountsProofOfOwnership: Sendable, FeatureReducer {
+	@ObservableState
+	struct State: Sendable, Hashable {
+		let accountAddresses: [AccountAddress]
+		let dappMetadata: DappMetadata
+		var signature: SignProofOfOwnership.State
+
+		var accounts: Accounts = []
+
+		@Presents
+		var destination: Destination.State?
+
+		init(
+			accountAddresses: [AccountAddress],
+			dappMetadata: DappMetadata,
+			challenge: DappToWalletInteractionAuthChallengeNonce
+		) {
+			self.accountAddresses = accountAddresses
+			self.dappMetadata = dappMetadata
+			self.signature = .init(dappMetadata: dappMetadata, challenge: challenge)
+		}
+	}
+
+	typealias Action = FeatureAction<Self>
+
+	enum ViewAction: Sendable, Equatable {
+		case task
+		case continueButtonTapped
+	}
+
+	enum InternalAction: Sendable, Equatable {
+		case setAccounts(Accounts)
+	}
+
+	enum DelegateAction: Sendable, Equatable {
+		case provenOwnership([AccountAuthProof], SignedAuthChallenge)
+		case failedToGetAccounts
+		case failedToSign
+	}
+
+	@CasePathable
+	enum ChildAction: Sendable, Equatable {
+		case signature(SignProofOfOwnership.Action)
```

Comment:
i think `signing` or `gather_proof` is a better name than `signature`

Reply:
Renamed to `sign` to follow the naming use for the other reducer